### ISSUE TITLE
Error 1 when running Invoke-CMApplyDriverPackage.ps1 in Task Sequence

### ISSUE THREAD

**Issue by TheGunslingerRG** [MSEndpointMgr/ModernDriverManagement#193]
We've been running an older version (2.x) for a long time and have just upgraded to V 4.2.0. Unfortunately, it isn't working with no clear understanding of why not. When re run the script as part of a task sequence, the step fails with error code 1.

If we run it in debug on a machine (that isn't the same make/model) and define the make/model as the examples, we see that it is able to detect the make/model and drivers successfully when we look at the logs at C:\Windows\Temp\ApplyDriverPacakge.log.

`\Invoke-CMApplyDriverPackage.ps1 -DebugMode -Endpoint "SiteURL" -UserName "username" -Password "password" -TargetOSVersion 21H2 -Manufacturer "Dell" -systemSKU "0A53" -TargetOSName "Windows 10" -ComputerModel "7090"`

We have verified that the admin service is working, that the account credentials are correct, and that the make/model exist in SCCM. We are defining the TargetOSName as well, which is new to us. Here's the command we're running, if it helps.

`-BareMetal -EndPoint "siteURL" -TargetOSName "Windows 10" -TargetOSVersion "21H2"`

Any ideas or are there ways to troubleshoot this? I do agree with a different poster that the download file for 4.2.0 is different than the raw 4.2.0 view in GitHub so we're using the version we copied from the Raw view (the download version threw a bunch of different errors).

**Comment by TheGunslingerRG** [comment]
I was finally able to get into WinPE and run the script manually to see any output and it simply gives a runtime error so not a whole lot of information there.

**Comment by TheGunslingerRG** [comment]
So I figured out the issue. Once I found the ApplyDriverPackage.log file in WinPE, it was a relatively easy series of issues to troubleshoot. Not sure if this is documented elsewhere but that path is X:\Windows\temp\SMSTSlog\. Once I discovered that, we found that we had models that didn't have current driver packages, we had models OS Version issues with the driver packages, etc. We resolved those and are imaging with v. 4.2.0 successfully.
